game/src/edit/lanes.rs

```diff
@@ -116,24 +116,19 @@ impl State<App> for LaneEditor {
         // Restrict what can be selected.
         if ctx.redo_mouseover() {
             app.recalculate_current_selection(ctx);
-            if let Some(ID::Lane(l)) = app.primary.current_selection {
-                if !can_edit_lane(&self.mode, l, app) {
-                    app.primary.current_selection = None;
+            if match app.primary.current_selection {
```

Comment:
I'm not sure if this rewrite is any more clear, or how to express things like this better.